src/browser/GooglePlusProxy.js

```diff
@@ -0,0 +1,141 @@
+                    "userId": profile.getId(),
+                    "displayName": profile.getName(),
+                    "familyName": profile.getFamilyName(),
+                    "givenName": profile.getGivenName(),
+                    "imageUrl": profile.getImageUrl()
+                });
+            }
+
+        } else {
+            if (error) error({'error': 'User not logged in.'});
+        }
+    },
+
+    trySilentLogin: function (success, error, options) {
+        if (!__googleSdkReady) {
+            return __googleCallbacks.push(function() {
+                this.trySilentLogin(success, error, options);
+            });
+        }
+
+        GooglePlusProxy.updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get(), success, error);
+    },
+
+    login: function (success, error, options) {
+        var that = this;
+        if (!__googleSdkReady) {
+            return __googleCallbacks.push(function() {
+                that.login(success, error, options);
+            });
+        }
+
+        gapi.auth2.getAuthInstance().signIn(options).then(function () {
+            GooglePlusProxy.updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get(), success, error);
+        }, function(err) {
+            error(err);
+        });
+    },
+
+    logout: function (success, error) {
+        if (!__googleSdkReady) {
+            return __googleCallbacks.push(function() {
+                this.logout(success, error);
+            });
+        }
+
+        gapi.auth2.getAuthInstance().signOut().then(success, function(err) {
+            error(err);
+        });
+    },
+
+    disconnect: function (success, error) {
+        if (!__googleSdkReady) {
+            return __googleCallbacks.push(function() {
+                this.disconnect(success, error);
+            });
+        }
+
+        gapi.auth2.getAuthInstance().disconnect().then(success, function(err) {
+            error(err);
+        });
+    },
+
+    getSigningCertificateFingerprint: function (success, error) {
+        console.warn('Not implemented.');
+        console.trace();
+    }
+};
+
+if (window.location.protocol === "file:") {
+    console.warn("Google API is not supported when using file:// protocol");
+} else {
+    window.handleClientLoad = function() {
+        gapi.load('auth2', function () {
+            gapi.auth2.init({
+                client_id: "WEB_APPLICATION_CLIENT_ID" // CLIENT_ID is populated by the cordova after_prepare hook
+            }).then(function () {
+                __googleSdkReady = true;
+
+                for (var i = 0; i < __googleCallbacks.length; i++) {
+                    __googleCallbacks[i].call(GooglePlusProxy);
+                }
+
+                // Listen for sign-in state changes.
+                gapi.auth2.getAuthInstance().isSignedIn.listen(GooglePlusProxy.updateSigninStatus);
+            }, function(error) {
+                if (error.details) {
+                    console.error(error.details);
```

Comment:
This always trigger this error for me, even though `WEB_APPLICATION_CLIENT_ID` is set in `config.xml`.

> Not a valid origin for the client: http://localhost:8000 has not been whitelisted for client ID null. Please go to https://console.developers.google.com/ and whitelist this origin for your project's client ID.